### PR TITLE
Add NetworkWatcher Extension testcase.

### PR DESCRIPTION
**Pull request by vakaranamsft**
This PR is to onboard Azure Network watcher extension onto LISA with initial test-case for extension validation on different Linux  flavors.

### CODE REVIEW

File: microsoft/testsuites/vm_extensions/networkwatcherextension.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
+from typing import Any
```
Comment (LiliDeng) [inline microsoft/testsuites/vm_extensions/networkwatcherextension.py:1]: one more blank line after this line

File: microsoft/testsuites/vm_extensions/networkwatcherextension.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+from lisa.operating_system import (
+    SLES,
+    AlmaLinux,
```
Comment (LiliDeng) [inline microsoft/testsuites/vm_extensions/networkwatcherextension.py]: remove unused  AlmaLinux

File: microsoft/testsuites/vm_extensions/networkwatcherextension.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        Installs and runs the Azure Network Watcher VM Extension.
+        Deletes the VM Extension.
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            supported_features=[AzureExtension],
+        ),
+    )
+    def verify_azure_network_watcher(self, log: Logger, node: Node) -> None:
+
```
Comment (LiliDeng) [inline microsoft/testsuites/vm_extensions/networkwatcherextension.py]: remove the blank line

File: microsoft/testsuites/vm_extensions/networkwatcherextension.py (hunk 1 above)
Comment (vakaranamsft) [reply microsoft/testsuites/vm_extensions/networkwatcherextension.py:1]: Added extra line

File: microsoft/testsuites/vm_extensions/networkwatcherextension.py (hunk 2 above)
Comment (vakaranamsft) [reply microsoft/testsuites/vm_extensions/networkwatcherextension.py]: Removed.

File: microsoft/testsuites/vm_extensions/networkwatcherextension.py (hunk 3 above)
Comment (vakaranamsft) [reply microsoft/testsuites/vm_extensions/networkwatcherextension.py]: Removed.

**Review by LiliDeng — APPROVED** [review]

File: microsoft/testsuites/vm_extensions/networkwatcherextension.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        # Delete VM Extension
+        extension.delete("AzureNetworkWatcherAgentLinux")
+
+        assert_that(
+            extension.check_exist("AzureNetworkWatcherAgentLinux")
+        ).described_as(
+            "Found the VM Extension still unexpectedly exists on the VM after deletion"
+        ).is_false()
+
+    def _is_supported_linux_distro(self, node: Node) -> bool:
```
Comment (squirrelsc) [inline microsoft/testsuites/vm_extensions/networkwatcherextension.py:75]: @LiliDeng This logic is used by multiple extensions, please create a task to share the logic in extension folder.

Comment (LiliDeng) [reply microsoft/testsuites/vm_extensions/networkwatcherextension.py:75]: > @LiliDeng This logic is used by multiple extensions, please create a task to share the logic in extension folder.

Ack

**Pull request merged**